src/targets/browser/launcher.ts

```diff
@@ -189,7 +192,13 @@ export async function attach(
     return new CdpConnection(connectionTransport, logger, telemetryReporter);
   } else if (browserURL) {
     const connectionURL = await retryGetWSEndpoint(browserURL, cancellationToken);
-    const connectionTransport = await WebSocketTransport.create(connectionURL, cancellationToken);
+
+    const inspectWs =
+      options.inspectUri && options.pageURL
```

Comment:
Technically we don't require a page URL in the general sense to use constructInspectorWSUri

```suggestion
      options.inspectUri
```